pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/MLPendingAckStoreProvider.java

```diff
@@ -159,15 +160,21 @@ public void openLedgerFailed(ManagedLedgerException exception, Object ctx) {
                                         pendingAckStoreFuture.completeExceptionally(exception);
                                     }
                                 }, () -> true, null);
-            });
-        }).exceptionally(e -> {
-            log.error("Failed to obtain the existence of ManagerLedger with topic and subscription : "
-                    + originPersistentTopic.getSubscriptions() + "  "
-                    + subscription.getName());
-            pendingAckStoreFuture.completeExceptionally(
-                    e.getCause());
-            return null;
-        });
+                    }).exceptionally(e -> {
+                        Throwable t = FutureUtil.unwrapCompletionException(e);
+                        log.error("[{}] [{}] Failed to get managedLedger config when init pending ack store!",
+                                originPersistentTopic, subscription, t);
+                        pendingAckStoreFuture.completeExceptionally(t);
+                        return null;
+
+                    });
+                }).exceptionally(e -> {
```

Comment:
Suggestion use `thenCompose` instead of `thenAccept`; this can save an `exceptionally` logic , just like this: 

```java
originPersistentTopic.getBrokerService().getManagedLedgerFactory()
                .asyncExists(TopicName.get(pendingAckTopicName)
                        .getPersistenceNamingEncoding()).thenCompose(exist -> {
            ...

            return originPersistentTopic.getBrokerService()
                    .getManagedLedgerConfig(topicName)

             ...

        }).exceptionally(e -> {
            log.error("Failed to obtain the existence of ManagerLedger with topic and subscription : "
                    + originPersistentTopic.getSubscriptions() + "  "
                    + subscription.getName());
            pendingAckStoreFuture.completeExceptionally(
                    e.getCause());
            return null;
        });
```

Reply:
If we use `thenCompose`, we cannot distinguish the exceptions thrown by check exist and get config, so we cannot print the corresponding log.

Reply:
ok